README.md

```diff
@@ -91,6 +92,74 @@ try {
 }
 ```
 
+## Conversion 2.x.x to 3.x.x
+> NOTE: this feature is still WIP, and is until the final release of 3.0.0
+
+Conversion to version `3.x.x` from `2.x.x` has several assumptions that should be know before converting:
+
+- The input must be valid AsyncAPI document.
+- External references are not resolved and converted, they remain untouched, even if they are incorrect.
+- In version `3.0.0`, the channel identifier is no longer its address, but due to the difficulty of defining a unique identifier, we still treat the address as an identifier. If there is a need to assign an identifier other than an address, an `x-channelId` extension should be defined at the level of the given channel.
+
+  ```yaml
+  # 2.x.x
+  channels:
+    users/signup:
+      x-channelId: 'userSignUp'
+      ...
+    users/logout:
+      ...
+
+  # 3.0.0
+  channels:
+    userSignUp:
+      ...
+    users/logout:
+      ...
+  ```
+
+- The `publish` operation is treated as a `receive` action, and `subscribe` is treated as a `send` action. Conversion by default is embraced from the application perspective. If you want to change this logic, you need to specify `v2tov3.pointOfView` configuration as `client`.
+- If the operation does not have an `operationId` field defined, the unique identifier of the operation will be defined as a combination of the identifier of the channel on which the operation was defined + the type of operation, `publish` or `subscribe`. Identical situation is with messages. However, here the priority is the `messageId` field and then the concatenation `{publish|subscribe}.messages.{optional index of oneOf messages}`.
+
+  ```yaml
+  # 2.x.x
+  channels:
+    users/signup:
+      publish:
+        message:
+          ...
+      subscribe:
+        operationId: 'userSignUpEvent'
+        message:
+          oneOf:
+            - messageId: 'userSignUpEventMessage'
+              ...
+            - ...
+        
+
+  # 3.0.0
+  channels:
+    users/signup:
+      messages:
+        publish.message:
+          ...
+        userSignUpEventMessage:
+          ...
+        userSignUpEvent.message.1:
+          ...
+  operations:
+    users/signup.publish:
+      action: receive
+      ...
+    userSignUpEvent:
+      action: send
+      ...
+  ```
+
+- Security requirements that use scopes are defined in the appropriate places inline, the rest as a reference to the `components.securitySchemes` objects.
+- If servers are defined at the channel level, they are converted as references to the corresponding objects defined in the `servers` field.
+- Channels and servers defined in components are also converted (unless configured - see [examples](#in-js)).
```

Comment:
You should add example in JS with options to skip `channels` and `servers` conversion.

Reply:
So maybe remove info about examples from this list item? And add after release 3.0.0.

Reply:
Yep, removed 👍